library/core/src/slice/mod.rs

```diff
@@ -1439,7 +1354,7 @@ impl<T> [T] {
     where
         F: FnMut(&T) -> bool,
     {
-        Split { v: self, pred, finished: false }
+        Split::new(self, pred, false)
```

Comment:
finish is always false here, so it probably makes sense to me this into `Split::new`

Reply:
Nice.